plugins/basestate/BaseState.cc

```diff
@@ -0,0 +1,243 @@
+                return;
+            }
+            if (!driver_properties.check("baseLink"))
+            {
+                yError() << "gz-sim-yarp-basestate-system : missing baseLink parameter";
+                return;
+            }
+            yInfo() << "gz-sim-yarp-basestate-system: configuration of device "
+                    << driver_properties.find("yarpDeviceName").asString()
+                    << " loaded from yarpConfigurationString : " << configuration_string << "\n";
+        } else
+        {
+            yError() << "gz-sim-yarp-basestate-system : missing yarpConfigurationString element";
+            return;
+        }
+
+        std::string deviceName = driver_properties.find("yarpDeviceName").asString();
+        std::string baseLinkName = driver_properties.find("baseLink").asString();
+
+        auto model = Model(_entity);
+
+        Entity baseLinkEntity = model.LinkByName(_ecm, baseLinkName);
+        if (_ecm.HasEntity(baseLinkEntity))
+        {
+            m_baseLinkEntity = baseLinkEntity;
+            yInfo() << "gz-sim-yarp-basestate-system: base link name '" << baseLinkName
+                    << "' found.";
+        } else
+        {
+            yError() << "gz-sim-yarp-basestate-system: base link name '" << baseLinkName
+                     << "' configured was not found in the model definition.";
+            return;
+        }
+
+        // Create link from entity
+        m_baseLink = Link(m_baseLinkEntity);
+        // Enable velocity computation in Gazebo
+        m_baseLink.EnableVelocityChecks(_ecm, true);
+        // Enable acceleration computation in Gazebo
+        m_baseLink.EnableAccelerationChecks(_ecm, true);
+
+        m_baseLinkScopedName = scopedName(m_baseLinkEntity, _ecm);
+        m_baseStateData.baseLinkScopedName = m_baseLinkScopedName;
+
+        driver_properties.put(YarpBaseStateScopedName.c_str(), m_baseLinkScopedName.c_str());
+
+        if (!driver_properties.check("yarpDeviceName"))
+        {
+            yError() << "gz-sim-yarp-basestate-system : missing yarpDeviceName parameter for "
+                        "device "
+                     << m_baseLinkScopedName;
+            return;
+        }
+
+        // Insert the pointer in the singleton handler for retrieving it in the yarp driver
+        BaseStateDataSingleton::getBaseStateDataHandler()->setBaseStateData(&(m_baseStateData));
+
+        driver_properties.put("device", "gazebo_basestate");
+        driver_properties.put("robot", m_baseLinkScopedName);
+
+        if (!m_baseStateDriver.open(driver_properties))
+        {
+            yError() << "gz-sim-yarp-basestate-system Plugin failed: error in opening yarp "
+                        "driver";
+            return;
+        }
+
+        m_deviceScopedName
+            = m_baseLinkScopedName + "/" + driver_properties.find("yarpDeviceName").asString();
+
+        if (!Handler::getHandler()->setDevice(m_deviceScopedName, &m_baseStateDriver))
```

Comment:
I opened this https://github.com/robotology/gz-sim-yarp-plugins/issues/69 to avoid forgetting.

Reply:
Ok thanks!